### PR TITLE
feat: add get command for ODH/RHOAI resources

### PR DESCRIPTION
**Pull request by ShettyGaurav**
## Description

Adds `kubectl odh get` command that provides simplified access to ODH/RHOAI custom resources without requiring users to know CRD group/version details.

**Supported resource types:**

| Resource | Alias | Kind |
|----------|-------|------|
| `notebooks` | `nb` | Kubeflow Notebook |
| `inferenceservices` | `isvc` | KServe InferenceService |
| `servingruntimes` | `sr` | KServe ServingRuntime |
| `datasciencepipelinesapplications` | `pipeline` | Data Science Pipelines |

**Features:**
- Resource registry with canonical names and short aliases
- Table output with ODH-specific columns per resource type (image, ready status, URL, age, etc.)
- JSON and YAML output formats (`-o json`, `-o yaml`)
- Standard kubectl flags: `-n`, `-A`, `-l`
- Graceful handling of missing CRDs, 403 Forbidden, and not-found errors
- Tab completion for resource type names

**New files:**
- `pkg/get/registry.go` — resource name/alias resolution
- `pkg/get/command.go` — command lifecycle (AddFlags/Complete/Validate/Run)
- `pkg/get/output.go` — table column definitions and output formatters
- `cmd/get/get.go` — Cobra wrapper
- `pkg/get/command_test.go` — unit tests with fake dynamic clients

**Modified files:**
- `cmd/main.go` — wire in the get subcommand
- `pkg/util/jq/jq.go` — defensive nil guard for typed nil pointers

## Usage
```
# List notebooks in current namespace
kubectl odh get nb

# List notebooks across all namespaces
kubectl odh get nb -A

# Get a specific notebook
kubectl odh get nb my-notebook -n my-project

# List inference services with label selector
kubectl odh get isvc -l app=my-model

# List serving runtimes
kubectl odh get sr -n my-project

# List pipelines in JSON format
kubectl odh get pipeline -o json

# Get a specific resource in YAML
kubectl odh get nb my-notebook -n my-project -o yaml
```
## Testing

- [X] Unit tests pass (`make test`)
- [X] Linter passes (`make lint`)
- [X] New functionality includes tests

## Review checklist

- [X] Code follows project conventions (see docs/coding/)

## Summary by CodeRabbit

* **New Features**
  * Added a new `get` command to list and retrieve resources with namespace filtering and kubectl-style persistent flags
  * Support for table, JSON, and YAML outputs and shell completion for resource names

* **Tests**
  * Added comprehensive tests covering resolution, listing, retrieval, output formats, and edge cases

* **Bug Fixes**
  * Fixed nil-handling for unstructured conversions to avoid panics

### CODE REVIEW

File: pkg/get/command.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+	if err != nil {
+		if meta.IsNoMatchError(err) {
+			c.IO.Fprintf(msgCRDNotInstalled, c.ResolvedType.Kind)
+
+			return nil
+		}
+
+		if apierrors.IsNotFound(err) {
+			return fmt.Errorf("%s %q not found in namespace %q", c.ResolvedType.Kind, c.ItemName, c.Namespace)
+		}
```
[the remark this answers is not in the document]
Comment (asanzgom) [reply pkg/get/command.go:191]: Line 173: fmt.Errorf("%s %q not found in namespace %q", ..., c.Namespace) — when AllNamespaces is true, c.Namespace is "", producing not found in namespace "". Should conditionally omit the namespace clause.

File: pkg/get/command.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+		return fmt.Errorf("getting %s %q: %w", c.ResolvedType.Kind, c.ItemName, err)
+	}
```
[the remark this answers is not in the document]
Comment (asanzgom) [reply pkg/get/command.go]: listResources (line 142) uses client.IsResourceTypeNotFound(err), but getResource (line 167) uses meta.IsNoMatchError(err). The dynamic client may not go through the REST mapper, so       
  meta.IsNoMatchError won't fire for missing CRDs on Get — it'll fall through to apierrors.IsNotFound, giving a confusing "not found in namespace" message instead of "CRD not installed".
                                                                                                                                                                                              
  Should use client.IsResourceTypeNotFound(err) in both paths. The ambiguity with IsNotFound (CRD missing vs resource missing) can be handled by keeping the apierrors.IsNotFound fallback    
  after the IsResourceTypeNotFound check — which is what listResources already does implicitly (list returns empty, not 404, when no resources exist).

[the remark this answers is not in the document]
Comment (asanzgom) [reply pkg/get/command.go]: 3. Missing test for getResource not-found path                                                                                                                                              
  
There's no test for the case where ItemName is set, but the resource doesn't exist. This would have caught this issue

**Review by asanzgom — APPROVED** [review]
